src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -9734,10 +9737,60 @@ Part::TopoShape SketchObject::getEdge(const Part::Geometry *geo, const char *nam
     return shape;
 }
 
-Data::IndexedName SketchObject::checkSubName(const char *sub) const
-{
-    // FIXME: trivial implementation needs to be replaced with full logic
-    (void)sub;
+Data::IndexedName SketchObject::checkSubName(const char *sub) const{

```

Comment:
I don't love the variable names in this method (too many abbreviations and similar-sounding-but-not-the-same things). @Pesc0 you want to take a stab at renaming some of them while you're at it? My linter also yells about the pointer arithmetic, so I wouldn't be sad to see that eliminated (though we want to be careful about efficiency here, I bet).